### PR TITLE
Added function overloading JSDoc for `add` function

### PR DESCRIPTION
**Pull request by goldenratio**
Hi,

@englercj  please check 

out of topic: `Class<T>`
For me `Class<async>` doesn't generate `typeof async`. I think you need to make new release for https://github.com/englercj/tsd-jsdoc

### CODE REVIEW

**Comment by englercj** [comment]
> For me Class<async> doesn't generate typeof async. I think you need to make new release for 

I do, I used my local version to generate the typings.

File: src/Loader.js [hunk 1]
```diff
@@ -280,9 +280,24 @@ export class Loader {
...
      *     .add('http://...', { crossOrigin: true }, function () {});
      * ```
-     *
-     * @param {string|IAddOptions} [name] - The name of the resource to load, if not passed the url is used.
-     * @param {string} [url] - The url for this resource, relative to the baseUrl of this loader.
+     */
+    /**
+     * @function
+     * @variation 1
+     * @param {...Array<any>} params
```
Comment (englercj) [inline src/Loader.js]: Could these `any`s be `IAddOptions` instead?

Comment (goldenratio) [reply src/Loader.js]: Doesn't seem to work without `any`s.
Btw, these signatures are based on pixi.d.ts

Comment (goldenratio) [reply src/Loader.js]:
```
add(param: string[] | IAddOptions[]): this;
  add(obj: string | string[] | IAddOptions | IAddOptions[], options?: IAddOptions, cb?: Loader.OnCompleteSignal): this;
  add(name: string, url: string, options?: IAddOptions, cb?: Loader.OnCompleteSignal): this;

```

what about this?
I guess we have to drop rest param,  I this case.

Comment (englercj) [reply src/Loader.js]: I'd prefer to drop the rest param. The overloads I'd like to see are the same as the documentation example above:

```ts
add(key: string, url: string, callback?: Loader.OnCompleteSignal): this;
add(key: string, url: string, options?: IAddOptions, callback?: Loader.OnCompleteSignal): this;
add(url: string, callback?: Loader.OnCompleteSignal): this;
add(url: string, options?: IAddOptions, callback?: Loader.OnCompleteSignal): this;
add(options: IAddOptions, callback?: Loader.OnCompleteSignal): this;
add(resources: (IAddOptions|string)[], callback?: Loader.OnCompleteSignal): this;
```

Comment (goldenratio) [reply src/Loader.js]: looks good.. I will update the PR

**Comment by englercj** [comment]
Went ahead and added this to master because I had a few minutes before work. Thanks for helping me figure this out!

**Comment by goldenratio** [comment]
@englercj Thanks. Any plans for MiniSignal types? :D

**Comment by goldenratio** [comment]
looks good. If any issues, I will let you know... Thanks a lot. 
Happy holidays!

**Comment by englercj** [comment]
Same to you!

**Pull request closed without merging**